pallets/pool-xyk/src/migrations/v3.rs

```diff
@@ -79,147 +91,147 @@ where
             }
         });
 
-        if let Err(_) = Pallet::<T>::withdraw_liquidity_unchecked(
+        Pallet::<T>::withdraw_liquidity_unchecked(
             user_account.clone(),
             dex_id,
             base_asset,
             target_asset,
             lp_tokens,
             1,
             1,
-        ) {
-            info!("Error encountered during liquidity withdrawal for account {:?} in pool {:?}, skipping", user_account, pool_account);
-            (
-                T::DbWeight::get()
-                    .reads_writes(2, 2)
-                    .saturating_add(<T as Config>::WeightInfo::withdraw_liquidity()),
-                false,
-            )
-        } else {
-            (
-                T::DbWeight::get()
-                    .reads_writes(3, 3)
-                    .saturating_add(<T as Config>::WeightInfo::withdraw_liquidity()),
-                true,
-            )
-        }
+        )?;
+
+        Ok(())
     }
 
     /// Removes corresponding entries from Properties, Reserves and TotalIssuances
     /// Also deregisters pool account and fee account from technical pallet
     fn remove_pool(
+        weight_meter: &mut WeightMeter,
         dex_id: T::DEXId,
         base_asset: T::AssetId,
         target_asset: T::AssetId,
         pool_account: T::AccountId,
-    ) -> Weight {
+    ) -> DispatchResult {
+        weight_meter.check_accrue(T::DbWeight::get().reads_writes(4, 8));
+
         let (_, tech_acc_id) =
-            Pallet::<T>::tech_account_from_dex_and_asset_pair(dex_id, base_asset, target_asset)
-                .unwrap();
+            Pallet::<T>::tech_account_from_dex_and_asset_pair(dex_id, base_asset, target_asset)?;
 
         let pair = TradingPair::<T::AssetId> {
             base_asset_id: base_asset.clone(),
             target_asset_id: target_asset.clone(),
         };
 
         EnabledSources::<T>::mutate(&dex_id, &pair, |opt_set| {
-            opt_set
-                .as_mut()
-                .unwrap()
-                .remove(&LiquiditySourceType::XYKPool)
+            if let Some(sources) = opt_set.as_mut() {
+                sources.remove(&LiquiditySourceType::XYKPool);
+            }
         });
         Properties::<T>::remove(base_asset, target_asset);
 
-        let fee_acc_id = tech_acc_id.to_fee_account().unwrap();
+        let fee_acc_id = tech_acc_id
+            .to_fee_account()
+            .ok_or(crate::Error::<T>::FeeAccountIsInvalid)?;
 
-        technical::Pallet::<T>::deregister_tech_account_id(tech_acc_id).unwrap();
-        technical::Pallet::<T>::deregister_tech_account_id(fee_acc_id).unwrap();
+        technical::Pallet::<T>::deregister_tech_account_id(tech_acc_id)?;
+        technical::Pallet::<T>::deregister_tech_account_id(fee_acc_id)?;
```

Comment:
Does this account have some balance? Should we transfer it somewhere before unregister?

Reply:
This account should not have any balance or will have some dust after liquidity withdraw. Some time ago we decided to leave it as is

Reply:
Ok